logger/logger.go

```diff
@@ -23,27 +27,44 @@ type Logger interface {
 	Error(string)
 }
 
+var logLevels = map[string]level.Option{
+	"debug": level.AllowDebug(),
+	"info":  level.AllowInfo(),
+	"warn":  level.AllowWarn(),
+	"error": level.AllowError(),
+}
```

Comment:
I would suggest adding enumeration instead of plain strings for log levels. Also, add method for casting string to enumeration values so that it can be used from `main.go`.

Reply:
I'm struggling a bit with this one. 
I removed the enumeration in [mainflux/level.go](https://github.com/mainflux/mainflux/pull/402/files#diff-a8b26aac2e521bc9f1a99413d2713efaL1) in favor of [vendor/level.go](https://github.com/mainflux/mainflux/blob/b07c6ae26499758753de2b223fab747610dceef9/vendor/github.com/go-kit/kit/log/level/level.go#L178-L205).
It seemed go-kit/log had already provided this functionaily. 

You're able to do `level.InfoValue().String()` similarly to previously, `log.Info.String()` in order to use from `main.go`. 

Admittedly I do like the look of  `log.Info.String()` more, but this removes what would be duplicate code.

Regarding the map I have created, it is being used as a way to convert what would be the env-var logLevel (type string) into a function. Adding a level filter requires a function be passed in [vendor/level.go](https://github.com/mainflux/mainflux/blob/b07c6ae26499758753de2b223fab747610dceef9/vendor/github.com/go-kit/kit/log/level/level.go#L75) So this is how I was mapping that user input string into the function the vendor package needs. Ex: `l = level.NewFilter(l, level.AllowError())`

I can change it to:
```
var logLevels = map[string]level.Option{
	level.DebugValue().String(): level.AllowDebug(),
	level.InfoValue().String():  level.AllowInfo(),
	level.WarnValue().String():  level.AllowWarn(),
	level.ErrorValue().String(): level.AllowError(),
}
```
If that would be prefered.

@anovakovic01 / @drasko  Please let me know your thoughts and please let me know if I misunderstood what was being suggested here!

Reply:
Idea is to create our levels, independent of go-kit logger, i.e. create an abstraction/wrapper there, so that we can replace go-kit logger in the future - for example with [zap](https://github.com/uber-go/zap).

Reply:
Ah ok that makes sense, I will work towards that approach and remove the dependency on go-kits levels

Reply:
Thanks @mfinley3.

Reply:
@anovakovic01 I've made some changes and it now uses the level wrapper and steers clear of go-kit/level. Please let me know your thoughts on the changes!